pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -116,9 +135,21 @@ func (c *component) SupportedOSType() rmeta.OSType {
 func (c *component) Objects() ([]client.Object, []client.Object) {
 	var objs []client.Object
 	// If l7spec is provided render the required objects.
-	c.config.envoyConfigMap = c.envoyL7ConfigMap()
 	objs = append(objs, c.serviceAccount())
+
+	// If Web Application Firewall is enabled, we need WAF ruleset ConfigMap present.
+	if c.config.WafEnabled {
+		// the first ConfigMap exists within the calico-operator namespace and stays as provided in config
```

Comment:
calico-operator -> tigera-operator

Also, per my comment above, I think we need to take the tigera-operator version of this configmap out of this render logic and put it in the controller the same way we do here: https://github.com/tigera/operator/blob/ed7dc7288f55e3f8bbc5804f316afdb5e8e774cd/pkg/controller/installation/core_controller.go#L1133-L1134